doc/source/whatsnew/v2.3.0.rst

```diff
@@ -108,6 +108,8 @@ Performance improvements
 
 Bug fixes
 ~~~~~~~~~
+- Fixed bug in :meth:`Series.diff` and :meth:`algorithms.diff` allowing non-integer values for the ``periods`` argument. (:issue:`56607`)
```

Comment:
```suggestion
- Fixed bug in :meth:`Series.diff` allowing non-integer values for the ``periods`` argument. (:issue:`56607`)
```

One last comment. ``pandas.core.algorithms`` isn't part of the public API, so there's no need to mention it in the whatsnew.